### PR TITLE
Fixed @GuiceUI and @GuiceViewChangeListener

### PR DESCRIPTION
**Pull request by peergynt**
- stick the @GuiceUI annotation in the UI scope as specified in the Javadoc.
- fixed ViewChangeListener argument check
- added PARAMETER target for @AllKnownGuiceViews annotation so it can be used as a constructor parameter for instance.

More to come as time permits

### CODE REVIEW

File: src/test/java/com/vaadinÐguice/server/UIScopeTest.java [hunk 1]
```diff
@@ -1,193 +0,0 @@
-package com.vaadin.guice.server;
```
Comment (berndhopp) [inline src/test/java/com/vaadinÐguice/server/UIScopeTest.java]: why remove these tests?

File: src/main/java/com/vaadin/guice/server/VaadinModule.java [hunk 2]
```diff
@@ -39,6 +40,7 @@ public VaadinModule(SessionProvider sessionProvider, Set<Class<? extends View>>
     @Override
     protected void configure() {
         bindScope(UIScope.class, uiScoper);
+        bindScope(GuiceUI.class, uiScoper);
```
Comment (berndhopp) [inline src/main/java/com/vaadin/guice/server/VaadinModule.java:43]: the documentation was copied from vaadin-spring, but we should stay as close to spring as possible so I guess this is the right way to go. Although I don't really see a point in putting the UI itself in the UI-Scope.

File: src/main/java/com/vaadin/guice/server/ReflectionUtils.java [hunk 3]
```diff
@@ -76,7 +76,7 @@ private ReflectionUtils() {
 
         for (Class<?> viewChangeListenerClass : reflections.getTypesAnnotatedWith(GuiceViewChangeListener.class)) {
             checkArgument(
-                    Module.class.isAssignableFrom(viewChangeListenerClass),
```
Comment (berndhopp) [inline src/main/java/com/vaadin/guice/server/ReflectionUtils.java:79]: that's awkward, my bad

File: src/test/java/com/vaadinÐguice/server/UIScopeTest.java (hunk 1 above)
Comment (peergynt) [inline src/test/java/com/vaadinÐguice/server/UIScopeTest.java]: It looks like commit 6edd44b added a bogus folder under src/test/java/com. If you have a look at what's on github, there are 2 folders: vaadin/guice and vaadin�guice. The second one seems to be a mistake. That's what I removed.

File: src/main/java/com/vaadin/guice/server/VaadinModule.java (hunk 2 above)
Comment (peergynt) [inline src/main/java/com/vaadin/guice/server/VaadinModule.java:43]: I think that it makes sense to have the UI added in the UI-scope. A possible use case would be to have a specific UI implementation injected somewhere (or a provider for that specific UI). I would expect that UI instance to be the 'singleton' UI within the current UI scope, not a brand new instance.

File: src/test/java/com/vaadinÐguice/server/UIScopeTest.java (hunk 1 above)
Comment (berndhopp) [inline src/test/java/com/vaadinÐguice/server/UIScopeTest.java]: alright, got it

**Comment by berndhopp** [comment]
good work, thanks

**Pull request merged**